app/main/messaging.py

```diff
@@ -0,0 +1,392 @@
+import string
+import itertools
+from flask import request, make_response, current_app, url_for
+from flask.ext.rq import get_queue
+from . import main
+from .. import db
+from ..utils import geocode, upload_image, get_rq_scheduler
+from ..models import Agency, IncidentReport, Location, User
+from ..reports.forms import IncidentReportForm
+from datetime import datetime, timedelta
+import twilio.twiml
+from twilio.rest import TwilioRestClient
+
+
+@main.route('/report_incident', methods=['GET'])  # noqa
+def handle_message():
+    """Called by Twilio when a text message is received."""
+    body = str(request.values.get('Body')).lower().strip()
+    num_media = int(request.values.get('NumMedia'))
+    twilio_hosted_media_url = str(request.values.get('MediaUrl0')) \
+        if num_media > 0 else None
+    message_sid = str(request.values.get('MessageSid'))
+    phone_number = str(request.values.get('From'))
+
+    twiml = twilio.twiml.Response()
+
+    # Retrieve incident cookies
+    step = int(request.cookies.get('messagecount', 0))
+    vehicle_id = str(request.cookies.get('vehicle_id', ''))
+    agency_name = str(request.cookies.get('agency_name', ''))
+    license_plate = str(request.cookies.get('license_plate', ''))
+    duration = int(request.cookies.get('duration', 0))
+    description = str(request.cookies.get('description', ''))
+    location = str(request.cookies.get('location', ''))
+    picture_url = str(request.cookies.get('picture_url', ''))
+
+    if 'report' == body.lower():
+        # reset report variables/cookies
+        vehicle_id = ''
+        agency_name = ''
+        license_plate = ''
+        duration = 0
+        description = ''
+        location = ''
+        picture_url = ''
+
+        step = handle_start_report(step, twiml)
+
+    elif step == 1:
```

Comment:
Would suggest against use of magic numbers here (`1, 2, 3, etc.`, see [stack overflow page](http://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad)) - maybe define some constants that correspond to each step. e.g. `STEP_INIT`, `STEP_AGENCY_NAME`, `STEP_VEHICLE_ID`, etc. (or whatever works best for the respective data you're storing)